test/handlers/FlexVotingClientHandler.sol

```diff
@@ -0,0 +1,235 @@
+  }
+
+  function _remainingTokens() internal returns (uint128) {
+    return type(uint128).max - ghost_mintedTokens;
+  }
+
+  function proposalLength() public returns(uint256) {
+    return _proposals.length();
+  }
+
+  // TODO This always creates a new actor. Should it?
+  function deposit(
+    uint208 _amount
+  ) createActor maybeCreateVoter countCall("deposit") external {
+    vm.assume(_remainingTokens() > 0);
+    _amount = uint208(bound(_amount, 0, _remainingTokens()));
+
+    // Some actors won't have the tokens they need. This is deliberate.
+    if (_amount <= _remainingTokens()) {
+      token.exposed_mint(currentActor, _amount);
+      ghost_mintedTokens += uint128(_amount);
+    }
+
+    vm.startPrank(currentActor);
+    // TODO we're pre-approving every depositi, should we?
+    token.approve(address(flexClient), uint256(_amount));
+    flexClient.deposit(_amount);
+    vm.stopPrank();
+
+    ghost_depositSum += _amount;
+    ghost_accountDeposits[currentActor] += uint128(_amount);
+  }
+
+  // TODO we restrict withdrawals to addresses that have balances, should we?
+  function withdraw(
+    uint256 _userSeed,
+    uint208 _amount
+  )
+    useActor(_userSeed)
+    countCall("withdraw")
+    external
+  {
+    // TODO we limit withdrawals to the total amount deposited, should we?
+    //   instead we could limit the caller to withdraw some portion of its balance
+    //   or we could let the caller attempt to withdraw any uint208
+    _amount = uint208(bound(_amount, 0, ghost_accountDeposits[currentActor]));
+
+    vm.prank(currentActor);
+    flexClient.withdraw(_amount);
+
+    ghost_withdrawSum += _amount;
+    ghost_accountDeposits[currentActor] -= uint128(_amount);
+  }
+
+  function propose(
+    string memory _proposalName,
+    uint256 _seed
+  ) countCall("propose") external {
+    // Proposal will underflow if we're on the zero block
+    if (block.number == 0) vm.roll(1);
+    if (this.proposalLength() > 3) return;
+
+    // Create a proposal
+    bytes memory receiverCallData = abi.encodeWithSignature("mockReceiverFunction()");
+    address[] memory targets = new address[](1);
+    uint256[] memory values = new uint256[](1);
+    bytes[] memory calldatas = new bytes[](1);
+    targets[0] = address(receiver);
+    values[0] = 0; // No ETH will be sent.
+    calldatas[0] = receiverCallData;
+
+    // Submit the proposal.
+    vm.prank(msg.sender);
+    uint256 _id = governor.propose(targets, values, calldatas, _proposalName);
+    _proposals.add(_id);
+
+    // Roll the clock to get voting started.
+    vm.roll(governor.proposalSnapshot(_id) + 1);
+  }
+
+  // TODO restrict expressVote to addresses that deposited BEFORE proposal.
+  function expressVote(
+    uint256 _proposalId,
+    uint8 _support,
+    uint256 _userSeed
+  ) useVoter(_userSeed) countCall("expressVote") external {
+    vm.assume(_proposals.length() > 0);
+
+    // TODO should we allow people to try to vote with bogus support types?
```

Comment:
Fine line but I would say no. That the `expressVote` method reverts on an invalid vote type is a property that can/should be tested sufficiently in unit tests. Invariant tests should primarily be about ensuring the integrity of the system as it is used.